From 859b3aa3dc2af5b73145424c8290ce058f5d8346 Mon Sep 17 00:00:00 2001
From: John Ralls <jralls@ceridwen.us>
Date: Fri, 4 Oct 2013 11:45:47 -0700
Subject: [PATCH] Restore gl_GLIBC21 to configure; needed for libcharset

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 1dde875..d014238 100644
--- a/configure.ac
+++ b/configure.ac
@@ -878,6 +878,7 @@ AC_STRUCT_DIRENT_D_TYPE
 AM_LANGINFO_CODESET
 AC_CHECK_HEADERS([stddef.h stdlib.h string.h])
 AC_CHECK_FUNCS(setlocale)
+gl_GLIBC21
 
 # check additional type sizes
 AC_CHECK_SIZEOF(size_t)
-- 
1.8.3.rc0

