From e1dcd5c77d52db0033cafd23d84f392c4d9f457e Mon Sep 17 00:00:00 2001
From: Hamish Mackenzie <Hamish.K.Mackenzie@googlemail.com>
Date: Sun, 29 Sep 2013 05:06:53 +1300
Subject: [PATCH] Fix infinite loop in gdkevents.c _gdk_event_queue_find_first

---
 gdk/gdkevents.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/gdk/gdkevents.c b/gdk/gdkevents.c
index 327b543..e770f47 100644
--- a/gdk/gdkevents.c
+++ b/gdk/gdkevents.c
@@ -96,16 +96,16 @@ _gdk_event_queue_find_first (GdkDisplay *display)
     {
       GdkEventPrivate *event = tmp_list->data;
 
-      if (event->flags & GDK_EVENT_PENDING)
-        continue;
-
-      if (pending_motion)
-        return pending_motion;
+      if (!(event->flags & GDK_EVENT_PENDING))
+        {
+          if (pending_motion)
+            return pending_motion;
 
-      if (event->event.type == GDK_MOTION_NOTIFY && !display->flushing_events)
-        pending_motion = tmp_list;
-      else
-        return tmp_list;
+          if (event->event.type == GDK_MOTION_NOTIFY && !display->flushing_events)
+            pending_motion = tmp_list;
+          else
+            return tmp_list;
+        }
 
       tmp_list = g_list_next (tmp_list);
     }
-- 
1.7.12.4 (Apple Git-37)

