diff -c /Users/john/Documents/Lion/Gnucash-Build/Gnucash-svn/src/cairo-1.10.2/configure\~ /Users/john/Documents/Lion/Gnucash-Build/Gnucash-svn/src/cairo-1.10.2/configure
*** a/configure	Sat Dec 25 06:22:57 2010
--- b/configure	Sun Jan  8 11:11:05 2012
***************
*** 17236,17242 ****
  MAYBE_WARN="$MAYBE_WARN -erroff=E_ENUM_TYPE_MISMATCH_ARG \
  			-erroff=E_ENUM_TYPE_MISMATCH_OP"
  
! MAYBE_WARN="$MAYBE_WARN -fno-strict-aliasing -fno-common -flto"
  
  MAYBE_WARN="$MAYBE_WARN -Wp,-D_FORTIFY_SOURCE=2"
  
--- 17236,17245 ----
  MAYBE_WARN="$MAYBE_WARN -erroff=E_ENUM_TYPE_MISMATCH_ARG \
  			-erroff=E_ENUM_TYPE_MISMATCH_OP"
  
! MAYBE_WARN="$MAYBE_WARN -fno-strict-aliasing -fno-common"
! if [ $CC != "llvm-gcc-4.2"]; then
!   MAYBE_WARN="$MAYBE_WARN -flto"
! fi
  
  MAYBE_WARN="$MAYBE_WARN -Wp,-D_FORTIFY_SOURCE=2"
  

Diff finished.  Sun Jan  8 11:12:32 2012
