diff -ruN a/pango/break.c b/pango/break.c
--- a/pango/break.c	2019-10-25 04:45:11.000000000 +0100
+++ b/pango/break.c	2020-12-01 20:28:47.000000000 +0000
@@ -405,7 +405,7 @@
                 GB_type = GB_Prepend;
                 break;
               }
-	    /* fall through */
+            __attribute__((fallthrough));
 	  case G_UNICODE_CONTROL:
 	  case G_UNICODE_LINE_SEPARATOR:
 	  case G_UNICODE_PARAGRAPH_SEPARATOR:
@@ -421,7 +421,7 @@
 		GB_type = GB_ControlCRLF;
 		break;
 	      }
-	    /* fall through */
+            __attribute__((fallthrough));
 
 	  case G_UNICODE_OTHER_LETTER:
 	    if (makes_hangul_syllable)
@@ -594,7 +594,7 @@
 		case G_UNICODE_CONTROL:
 		  if (wc != 0x000D && wc != 0x000A && wc != 0x000B && wc != 0x000C && wc != 0x0085)
 		    break;
-		  /* fall through */
+                  __attribute__((fallthrough));
 		case G_UNICODE_LINE_SEPARATOR:
 		case G_UNICODE_PARAGRAPH_SEPARATOR:
 		  WB_type = WB_NewlineCRLF; /* CR, LF, Newline */
diff -ruN a/pango/pango-renderer.c b/pango/pango-renderer.c
--- a/pango/pango-renderer.c	2019-10-25 04:45:11.000000000 +0100
+++ b/pango/pango-renderer.c	2020-12-01 20:29:56.000000000 +0000
@@ -222,7 +222,7 @@
 				     rect->y + 2 * rect->height,
 				     rect->width,
 				     rect->height);
-      /* Fall through */
+      __attribute__((fallthrough));
     case PANGO_UNDERLINE_SINGLE:
     case PANGO_UNDERLINE_LOW:
       pango_renderer_draw_rectangle (renderer,
diff -ruN a/tests/testboundaries_ucd.c b/tests/testboundaries_ucd.c
--- a/tests/testboundaries_ucd.c	2019-10-25 04:45:11.000000000 +0100
+++ b/tests/testboundaries_ucd.c	2020-12-01 20:31:12.000000000 +0000
@@ -104,7 +104,7 @@
         {
         case 0x00f7: /* DIVISION SIGN: boundary here */
           temp_attr.bits |= bits.bits;
-          /* fall through */
+          __attribute__((fallthrough));
 
         case 0x00d7: /* MULTIPLICATION SIGN: no boundary here */
           break;
