diff --git a/giscanner/scannerlexer.l b/giscanner/scannerlexer.l
index 6a40398..40f1fc0 100644
--- a/giscanner/scannerlexer.l
+++ b/giscanner/scannerlexer.l
@@ -108,6 +108,7 @@ stringtext				([^\\\"])|(\\.)
 "|"					{ return '|'; }
 "~"					{ return '~'; }
 "!"					{ return '!'; }
+"@"					{ return '@'; }
 "="					{ return '='; }
 "<"					{ return '<'; }
 ">"					{ return '>'; }
@@ -199,6 +200,7 @@ stringtext				([^\\\"])|(\\.)
 
 "\""{stringtext}*"\""			{ return STRING; }
 "L\""{stringtext}*"\""			{ return STRING; }
+"@\""{stringtext}*"\""			{ return STRING; }
 
 .					{ if (yytext[0]) fprintf(stderr, "%s:%d: unexpected character `%c'\n", scanner->current_filename, lineno, yytext[0]); }
 
