--- a/meta/guild.in	Fri Feb  1 09:19:34 2013
+++ b/meta/guild.in	Mon Dec 14 12:03:44 2015
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!@SHELL@
 # -*- scheme -*-
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
--- a/meta/Makefile.in
+++ b/meta/Makefile.in
@@ -1586,7 +1586,8 @@ substitute = \
 	 -e "s|[@]sitedir[@]|$(sitedir)|g"					\
 	 -e "s|[@]guile[@]|$$guile|g"						\
 	 -e "s|[@]guild[@]|$$guild|g"						\
-	 -e "s|[@]installed_guile[@]|$$installed_guile|g"
+	 -e "s|[@]installed_guile[@]|$$installed_guile|g"			\
+	 -e "s|[@]SHELL[@]|$$CONFIG_SHELL|g"
 
 
 # Substitutions for dependencies that appear in 'guile-2.0.pc'.
