diff -c /Users/john/Development/Gramps-Build/gramps-41-git-Leopard/src/nettle-2.7.1/Makefile.in\~ /Users/john/Development/Gramps-Build/gramps-41-git-Leopard/src/nettle-2.7.1/Makefile.in
--- a/Makefile.in	Tue May 28 07:21:53 2013
+++ b/Makefile.in	Sun Feb 22 15:31:18 2015
@@ -229,27 +229,27 @@
 # executable. Avoid object file targets to make it easy to run the
 # right compiler.
 aesdata$(EXEEXT_FOR_BUILD): aesdata.c
-	$(CC_FOR_BUILD) `test -f aesdata.c || echo '$(srcdir)/'`aesdata.c \
+	$(CC_FOR_BUILD) $(CFLAGS) `test -f aesdata.c || echo '$(srcdir)/'`aesdata.c \
 	-o aesdata$(EXEEXT_FOR_BUILD)
 
 desdata$(EXEEXT_FOR_BUILD): desdata.c
-	$(CC_FOR_BUILD) `test -f desdata.c || echo '$(srcdir)/'`desdata.c \
+	$(CC_FOR_BUILD) $(CFLAGS) `test -f desdata.c || echo '$(srcdir)/'`desdata.c \
 	-o desdata$(EXEEXT_FOR_BUILD)
 
 twofishdata$(EXEEXT_FOR_BUILD): twofishdata.c
-	$(CC_FOR_BUILD) `test -f twofishdata.c || echo '$(srcdir)/'`twofishdata.c \
+	$(CC_FOR_BUILD) $(CFLAGS) `test -f twofishdata.c || echo '$(srcdir)/'`twofishdata.c \
 	-o twofishdata$(EXEEXT_FOR_BUILD)
 
 shadata$(EXEEXT_FOR_BUILD): shadata.c
-	$(CC_FOR_BUILD) `test -f shadata.c || echo '$(srcdir)/'`shadata.c -lm \
+	$(CC_FOR_BUILD) $(CFLAGS) `test -f shadata.c || echo '$(srcdir)/'`shadata.c -lm \
 	-o shadata$(EXEEXT_FOR_BUILD)
 
 gcmdata$(EXEEXT_FOR_BUILD): gcmdata.c
-	$(CC_FOR_BUILD) `test -f gcmdata.c || echo '$(srcdir)/'`gcmdata.c \
+	$(CC_FOR_BUILD) $(CFLAGS) `test -f gcmdata.c || echo '$(srcdir)/'`gcmdata.c \
 	-o gcmdata$(EXEEXT_FOR_BUILD)
 
 eccdata$(EXEEXT_FOR_BUILD): eccdata.c mini-gmp.c mini-gmp.h
-	$(CC_FOR_BUILD) `test -f eccdata.c || echo '$(srcdir)/'`eccdata.c \
+	$(CC_FOR_BUILD) $(CFLAGS) `test -f eccdata.c || echo '$(srcdir)/'`eccdata.c \
 	-o eccdata$(EXEEXT_FOR_BUILD)
 
 # desCore rules

Diff finished.  Sun Feb 22 15:31:34 2015
