diff --git a/Source/WebKit2/WebProcess/EntryPoint/unix/WebProcessMain.cpp~ b/Source/WebKit2/WebProcess/EntryPoint/unix/WebProcessMain.cpp
index 480f9d0..26e351b 100644
--- a/Source/WebKit2/WebProcess/EntryPoint/unix/WebProcessMain.cpp~
+++ b/Source/WebKit2/WebProcess/EntryPoint/unix/WebProcessMain.cpp
@@ -23,6 +23,8 @@
  * THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <config.h>
+#include <wtf/MainThread.h>
 #include "WebProcessMainUnix.h"
 
 #include <cstdlib>
@@ -40,6 +42,6 @@ int main(int argc, char** argv)
     // https://bugzilla.gnome.org/show_bug.cgi?id=738633
     // WARNING: This needs to be KEPT IN SYNC with WebProcessMain.cpp.
     setenv("G_TLS_GNUTLS_PRIORITY", "NORMAL:%COMPAT:!VERS-SSL3.0:!ARCFOUR-128", 0);
-
+    WTF::initializeMainThread();
     return WebProcessMainUnix(argc, argv);
 }
