diff --git a/Source/WTF/wtf/text/StringView.h~ b/Source/WTF/wtf/text/StringView.h
index ef209f9..42ecca8 100644
--- a/Source/WTF/wtf/text/StringView.h~
+++ b/Source/WTF/wtf/text/StringView.h
@@ -37,7 +37,7 @@
 #include <wtf/text/StringCommon.h>
 
 // FIXME: Enabling the StringView lifetime checking causes the MSVC build to fail. Figure out why.
-#if defined(NDEBUG) || COMPILER(MSVC)
+#if defined(NDEBUG) || COMPILER(MSVC) || __APPLE__
 #define CHECK_STRINGVIEW_LIFETIME 0
 #else
 #define CHECK_STRINGVIEW_LIFETIME 1
