diff --git a/gdk/quartz/gdkscreen-quartz.c b/gdk/quartz/gdkscreen-quartz.c
index e83c87b..e28f3b2 100644
--- a/gdk/quartz/gdkscreen-quartz.c
+++ b/gdk/quartz/gdkscreen-quartz.c
@@ -208,7 +208,11 @@ _gdk_quartz_screen_update_window_sizes (GdkScreen *screen)
   windows = gdk_screen_get_toplevel_windows (screen);
 
   for (list = windows; list; list = list->next)
-    _gdk_quartz_window_update_position (list->data);
+    {
+      if (GDK_WINDOW_TYPE(list->data) == GDK_WINDOW_OFFSCREEN)
+        continue;
+      _gdk_quartz_window_update_position (list->data);
+    }
 
   g_list_free (windows);
 }
