From 259563958047ccbf6f61578f2d724fc731218304 Mon Sep 17 00:00:00 2001
From: John Ralls <jralls@ceridwen.us>
Date: Sun, 25 Sep 2011 12:03:54 -0700
Subject: [PATCH 08/15] Implement GtkDragSourceOwner pasteboardChangedOwner:

---
 gtk/gtkdnd-quartz.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/gtk/gtkdnd-quartz.c b/gtk/gtkdnd-quartz.c
index 084aada..21ce11a 100644
--- a/gtk/gtkdnd-quartz.c
+++ b/gtk/gtkdnd-quartz.c
@@ -149,6 +149,8 @@ struct _GtkDragFindData
   guint target_info;
   GtkSelectionData selection_data;
 
+  g_return_if_fail(info->source_widget != NULL);
+  g_return_if_fail(info->target_list != NULL);
   selection_data.selection = GDK_NONE;
   selection_data.data = NULL;
   selection_data.length = -1;
@@ -171,6 +173,15 @@ struct _GtkDragFindData
     }
 }
 
+- (void)pasteboardChangedOwner: (NSPasteboard*)sender
+{
+  if (!info) return;
+
+  info->target_list = NULL;
+  info->widget = NULL;
+  info->source_widget = NULL;
+}
+
 - (id)initWithInfo:(GtkDragSourceInfo *)anInfo
 {
   self = [super init];

