diff -u /Users/john/Development/gtk-build/gtk-stable-10.9-x86_64/src/cairomm-1.12.2/cairomm/enums.h\~ /Users/john/Development/gtk-build/gtk-stable-10.9-x86_64/src/cairomm-1.12.2/cairomm/enums.h
--- a/cairomm/enums.h	2015-08-12 02:27:20.000000000 -0700
+++ b/cairomm/enums.h	2017-04-30 14:08:29.000000000 -0700
@@ -20,7 +20,9 @@
 #define __CAIROMM_ENUMS_H
 
 #include <cairo.h>
+#ifdef CAIRO_HAS_FT_FONT
 #include <cairo-ft.h>
+#endif //CAIRO_HAS_FT_FONT
 
 namespace Cairo
 {
@@ -828,6 +830,7 @@
  *
  * @since 1.12
  */
+#ifdef CAIRO_HAS_FT_FONT
 enum FtSynthesize {
 
     /// Embolden the glyphs (redraw with a pixel offset)
@@ -846,7 +849,7 @@
 {
     return static_cast<FtSynthesize>(static_cast<int>(a) & static_cast<int>(b));
 }
-
+#endif //CAIRO_HAS_FT_FONT
 } // namespace Cairo
 
 #endif //__CAIROMM_ENUMS_H

Diff finished.  Sun Apr 30 15:22:11 2017
