--- a/test/Makefile.am	2018-08-16 18:10:53.000000000 -0700
+++ b/test/Makefile.am	2018-11-08 20:40:55.000000000 -0800
@@ -158,8 +158,10 @@
 svg-clip			\
 svg-surface			\
 toy-font-face			\
-font-variations			\
 user-data
+#if CAIRO_HAS_FT_FONT
+NOLOG_TESTS += font-variations
+#endif

 # A target to summarise the failures
 check-summary:
--- a/test/Makefile.sources	2018-08-16 18:10:53.000000000 -0700
+++ b/test/Makefile.sources	2018-11-08 20:39:26.000000000 -0800
@@ -144,7 +144,6 @@
 	font-face-get-type.c				\
 	font-matrix-translation.c			\
 	font-options.c					\
-	font-variations.c				\
 	glyph-cache-pressure.c				\
 	get-and-set.c					\
 	get-clip.c					\
@@ -399,5 +398,6 @@

 ft_font_test_sources = \
 	bitmap-font.c \
+	font-variations.c \
 	ft-font-create-for-ft-face.c \
 	ft-show-glyphs-positioning.c \
 	ft-show-glyphs-table.c \