diff --git a/src/enchant.c b/src/enchant.c
index 79fb21c..6f57fff 100644
--- a/src/enchant.c
+++ b/src/enchant.c
@@ -250,6 +250,11 @@ enchant_get_module_dirs (void)
 			module_dirs = enchant_slist_append_unique_path (module_dirs, module_dir);
 		}
 #endif
+	module_dir = g_getenv("ENCHANT_MODULE_PATH");
+	if (module_dir)
+		{
+			module_dirs = enchant_slist_append_unique_path (module_dirs, g_strdup(module_dir));
+		}
 
 	return module_dirs;
 }
@@ -259,6 +264,7 @@ enchant_get_conf_dirs (void)
 {
 	GSList *conf_dirs = NULL, *user_conf_dirs, *iter;
 	char * ordering_dir = NULL, * prefix = NULL;
+	char *xdg_data_dir;
 
 	user_conf_dirs = enchant_get_user_config_dirs();
 
@@ -291,6 +297,19 @@ enchant_get_conf_dirs (void)
 	conf_dirs = enchant_slist_append_unique_path (conf_dirs, g_strdup (ENCHANT_GLOBAL_ORDERING));
 #endif
 
+	xdg_data_dir = g_getenv ("XDG_DATA_DIRS");
+	if (xdg_data_dir)
+		{
+			char **xdg_ddirs = g_strsplit (xdg_data_dir, ":", 0);
+			char **xdg_iter;
+
+			for (xdg_iter = xdg_ddirs; *xdg_iter != NULL; xdg_iter++)
+				{
+					conf_dirs = enchant_slist_append_unique_path (conf_dirs, g_strdup(*xdg_iter));
+				}
+			g_strfreev (xdg_ddirs);
+		}
+
 	return conf_dirs;
 }
 
