diff -c /Users/john/Development/Gramps-Build/gramps-41-git-Leopard/src/freetype-2.5.3/builds/unix/unix-cc.in\~ /Users/john/Development/Gramps-Build/gramps-41-git-Leopard/src/freetype-2.5.3/builds/unix/unix-cc.in
--- a/builds/unix/unix-cc.in	Tue Aug 31 04:04:28 2010
+++ b/builds/unix/unix-cc.in	Sun Feb 22 14:14:44 2015
@@ -80,6 +80,7 @@
 #
 CPPFLAGS := @CPPFLAGS@
 CFLAGS   := -c @XX_CFLAGS@ @CFLAGS@ -DFT_CONFIG_CONFIG_H="<ftconfig.h>"
+CCexe_CFLAGS := @XX_CFLAGS@ @CFLAGS@ -DFT_CONFIG_CONFIG_H="<ftconfig.h>"
 
 # ANSIFLAGS: Put there the flags used to make your compiler ANSI-compliant.
 #
@@ -94,6 +95,7 @@
 # Linker flags.
 #
 LDFLAGS := @LDFLAGS@
+CCexe_LDFLAGS := @LDFLAGS@
 
 
 # export symbols

Diff finished.  Sun Feb 22 14:18:35 2015
