diff --git a/meson.build b/meson.build
index 97f309988..74fde432f 100644
--- a/meson.build
+++ b/meson.build
@@ -237,6 +237,20 @@ endif
 ft2_deps = []
 
 
+# Correct compatibility version for OS x
+#
+# OSX sets the compatibility_version (aka libtools version) differently from
+# the library name.
+#
+if host_machine.system() == 'darwin'
+  # Maintain compatibility with autotools on macOS
+  common_ldflags = [
+    '-compatibility_version', ft2_pkgconfig_version.split('.')[0],
+    '-current_version', ft2_pkgconfig_version
+  ]
+endif
+
+
 # Generate `ftoption.h` based on available dependencies.
 
 process_header_command = [python_exe,
@@ -344,6 +358,7 @@ ft2_lib = library('freetype',
   dependencies: ft2_deps,
   install: true,
   version: ft2_so_version,
+  link_args: common_ldflags,
 )
 
 
