diff --git a/app/main.c.orig b/app/main.c
index 5de0b30..27bf241 100644
--- a/app/main.c.orig
+++ b/app/main.c
@@ -384,7 +384,7 @@ gimp_macos_setenv (const char * progname)
       tmp = g_strdup_printf ("%s", res_dir);
       g_setenv ("PYTHONHOME", tmp, TRUE);
       g_free (tmp);
-      tmp = g_strdup_printf ("%s/lib/python2.7:%s/lib/gimp/2.0/python", res_dir, res_dir);
+      tmp = g_strdup_printf ("%s/lib/python3.9", res_dir);
       g_setenv ("PYTHONPATH", tmp, TRUE);
       g_free (tmp);
       tmp = g_strdup_printf ("%s/lib/gio/modules", res_dir);
@@ -393,6 +393,12 @@ gimp_macos_setenv (const char * progname)
       tmp = g_strdup_printf ("%s/share/libwmf/fonts", res_dir);
       g_setenv ("WMF_FONTDIR", tmp, TRUE);
       g_free (tmp);
+      if (g_getenv ("XDG_DATA_DIRS"))
+        tmp = g_strdup_printf ("%s/share:%s", res_dir, g_getenv ("XDG_DATA_DIRS"));
+      else
+        tmp = g_strdup_printf ("%s/share", res_dir);
+      g_setenv ("XDG_DATA_DIRS", tmp, TRUE);
+      g_free (tmp);
       if (g_getenv ("HOME") != NULL)
         {
           tmp = g_strdup_printf ("%s/Library/Application Support/GIMP/3.00/cache",
