diff -u /Users/john/Development/gtk-build/gtk-stable-10.15-x86_64/src/gnutls-3.6.12/configure.ac\~ /Users/john/Development/gtk-build/gtk-stable-10.15-x86_64/src/gnutls-3.6.12/configure.ac
--- a/configure.ac	2020-01-31 15:14:22.000000000 -0800
+++ b/configure.ac	2020-04-14 09:36:10.000000000 -0700
@@ -122,14 +122,6 @@
   ;;
   *darwin*)
     have_macosx=yes
-    save_LDFLAGS="$LDFLAGS"
-    dnl Try to use -no_weak_imports if available. This makes sure we
-    dnl error out when linking to a function that doesn't exist in the
-    dnl intended minimum runtime version.
-    LDFLAGS="$LDFLAGS -Wl,-no_weak_imports"
-    AC_MSG_CHECKING([whether the linker supports -Wl,-no_weak_imports])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([], [])],
-      [AC_MSG_RESULT(yes)], [AC_MSG_RESULT(no); LDFLAGS="$save_LDFLAGS"])
   ;;
   *solaris*)
     have_elf=yes

Diff finished.  Tue Apr 14 09:48:42 2020
