diff --git a/tests/Makefile.am b/tests/Makefile.am
index e011484..c34b72e 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -14,7 +14,7 @@ DEPS =									\
 	$(top_builddir)/gdk/$(gdktargetlib)				\
 	$(top_builddir)/gtk/$(gtktargetlib)
 
-LDADDS =								\
+LDADD =								\
 	$(top_builddir)/gdk/$(gdktargetlib)				\
 	$(top_builddir)/gtk/$(gtktargetlib)				\
 	$(GTK_DEP_LIBS)							\
@@ -166,72 +166,6 @@ testtooltips_DEPENDENCIES = $(TEST_DEPS)
 testvolumebutton_DEPENDENCIES = $(TEST_DEPS)
 testwindows_DEPENDENCIES = $(TEST_DEPS)
 
-flicker_LDADD = $(LDADDS)
-simple_LDADD = $(LDADDS)
-print_editor_LDADD = $(LDADDS)
-testaccel_LDADD = $(LDADDS)
-testassistant_LDADD = $(LDADDS)
-testbbox_LDADD = $(LDADDS)
-testbuttons_LDADD = $(LDADDS)
-testframe_LDADD = $(LDADDS)
-testcairo_LDADD = $(LDADDS)
-testcalendar_LDADD = $(LDADDS)
-testcombo_LDADD = $(LDADDS)
-testcombochange_LDADD = $(LDADDS)
-testcellrenderertext_LDADD = $(LDADDS)
-testclientmessage_LDADD = $(LDADDS)
-testdnd_LDADD = $(LDADDS)
-testentrycompletion_LDADD = $(LDADDS)
-testentryicons_LDADD = $(LDADDS)
-testfilechooser_LDADD = $(LDADDS)
-testfilechooserbutton_LDADD = $(LDADDS)
-testgtk_LDADD = $(LDADDS)
-testicontheme_LDADD = $(LDADDS)
-testiconview_LDADD = $(LDADDS)
-testiconview_keynav_LDADD = $(LDADDS)
-testinput_LDADD = $(LDADDS)
-testimage_LDADD = $(LDADDS)
-testmenus_LDADD = $(LDADDS)
-testmenubars_LDADD = $(LDADDS)
-testmountoperation_LDADD = $(LDADDS)
-testmultidisplay_LDADD = $(LDADDS)
-testmultiscreen_LDADD = $(LDADDS)
-testnotebookdnd_LDADD = $(LDADDS)
-testnouiprint_LDADD = $(LDADDS)
-testoffscreen_LDADD = $(LDADDS)
-testoffscreenwindow_LDADD = $(LDADDS)
-testorientable_LDADD = $(LDADDS)
-testprint_LDADD = $(LDADDS)
-testrecentchooser_LDADD = $(LDADDS)
-testrecentchoosermenu_LDADD = $(LDADDS)
-testrgb_LDADD = $(LDADDS)
-testrichtext_LDADD = $(LDADDS)
-testscale_LDADD = $(LDADDS)
-testselection_LDADD = $(LDADDS)
-testsocket_LDADD = $(LDADDS)
-testsocket_child_LDADD = $(LDADDS)
-testspinbutton_LDADD = $(LDADDS)
-teststatusicon_LDADD = $(LDADDS)
-testtoolbar_LDADD = $(LDADDS)
-stresstest_toolbar_LDADD = $(LDADDS)
-testtreeedit_LDADD = $(LDADDS)
-testtreemodel_LDADD = $(LDADDS)
-testtreeview_LDADD = $(LDADDS)
-testtreefocus_LDADD = $(LDADDS)
-testtreeflow_LDADD = $(LDADDS)
-testtreecolumns_LDADD = $(LDADDS)
-testtreecolumnsizing_LDADD = $(LDADDS)
-testtreesort_LDADD = $(LDADDS)
-testtext_LDADD = $(LDADDS)
-treestoretest_LDADD = $(LDADDS)
-testxinerama_LDADD = $(LDADDS)
-testmerge_LDADD = $(LDADDS)
-testactions_LDADD = $(LDADDS)
-testgrouping_LDADD = $(LDADDS)
-testtooltips_LDADD = $(LDADDS)
-testvolumebutton_LDADD = $(LDADDS)
-testwindows_LDADD = $(LDADDS)
-
 
 testentrycompletion_SOURCES = 	\
 	prop-editor.c		\
