--- a/gtk/gtkapplicationwindow.c	2015-07-21 07:22:36.000000000 -0700
+++ b/gtk/gtkapplicationwindow.c	2018-01-28 16:04:38.000000000 -0800
@@ -29,7 +29,7 @@
 #include "gtkintl.h"
 #include "gtksettings.h"
 
-#ifdef HAVE_GIO_UNIX
+#if !defined(__APPLE__) && defined(HAVE_GIO_UNIX)
 #include <gio/gdesktopappinfo.h>
 #endif
 
@@ -255,7 +255,7 @@
 {
   gchar *retval = NULL;
 
-#ifdef HAVE_GIO_UNIX
+#if !defined(__APPLE__) && defined(HAVE_GIO_UNIX)
   GDesktopAppInfo *app_info;
   const gchar *app_name = NULL;
   gchar *desktop_file;
--- a/gtk/gtk-launch.c	2014-11-26 18:54:55.000000000 -0800
+++ b/gtk/gtk-launch.c	2018-01-28 16:03:08.000000000 -0800
@@ -28,7 +28,7 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <gio/gio.h>
-#ifdef G_OS_UNIX
+#if !defined(__APPLE__) && defined(G_OS_UNIX)
 #include <gio/gdesktopappinfo.h>
 #endif
 #include <gtk.h>
@@ -109,7 +109,7 @@
   gtk_init (&argc, &argv);
 
   app_name = *args;
-#ifdef G_OS_UNIX
+#if !defined(__APPLE__) && defined(G_OS_UNIX)
   if (g_str_has_suffix (app_name, ".desktop"))
     desktop_file_name = g_strdup (app_name);
   else 
