diff -c /Users/john/Development/Gnucash-Build/Gnucash-master-git/src/guile-2.0.12/libguile/stime.c\~ /Users/john/Development/Gnucash-Build/Gnucash-master-git/src/guile-2.0.12/libguile/stime.c
--- a/libguile/stime.c	Mon Jul 11 09:19:59 2016
+++ b/libguile/stime.c	Mon Mar 13 14:02:59 2017
@@ -828,7 +828,7 @@
   if (clock_gettime (CLOCK_REALTIME, &posix_real_time_base) == 0)
     get_internal_real_time = get_internal_real_time_posix_timer;
 
-#ifdef HAVE_POSIX_CPUTIME
+#if defined(HAVE_POSIX_CPUTIME) && defined(HAVE_CPU_CLOCKID)
   {
     clockid_t dummy;
     

Diff finished.  Mon Mar 13 14:04:07 2017
