diff -c /Users/john/Development/gtk-sources/p11-kit-0.22.1/common/compat.c\~ /Users/john/Development/gtk-sources/p11-kit-0.22.1/common/compat.c
--- a/common/compat.c~	Sun Oct  5 22:29:32 2014
+++ b/common/compat.c	Fri Feb 20 12:48:53 2015
@@ -918,7 +918,11 @@
 
 			/* didn't parse or is the opendir() fd */
 			if (!end || *end != '\0' ||
+#if defined(__APPLE__)
+			    (int)num == dir->__dd_fd)
+#else
 			    (int)num == dirfd (dir))
+#endif
 				continue;
 
 			fd = num;

Diff finished.  Sun Feb 22 13:37:16 2015
