--- configure.ac.orig	2019-10-29 01:36:14.000000000 +0100
+++ configure.ac	2019-10-29 01:37:12.000000000 +0100
@@ -61,7 +61,7 @@
 # Initialize Automake with the following settings:
 # Relax some of the GNU requirements
 # Report all potential warnings and report all warnings as errors
-AM_INIT_AUTOMAKE([foreign -Wall -Werror])
+AM_INIT_AUTOMAKE([foreign -Wall])
 
 # Enable silent rules is available
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
@@ -84,7 +84,7 @@
 # the discussion at http://lists.gnu.org/archive/html/bug-autoconf/2010-01/msg00031.html
 
 # Pedantic errors please
-CFLAGS="$CFLAGS -Wall -ansi -pedantic"
+CFLAGS="$CFLAGS -Wall -std=gnu99 -pedantic"
 
 # Find the standard math functions
 # Taken from http://www.flameeyes.eu/autotools-mythbuster/autoconf/finding.html
@@ -94,7 +94,7 @@
 ])
 
 # Warnings as errors please
-CFLAGS="$CFLAGS -Werror"
+CFLAGS="$CFLAGS"
 
 # Find GLib support via pkg-config
 PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.28])
