--- a/src/wscript	2015-04-12 20:24:07.000000000 -0700
+++ b/src/wscript	2015-04-12 20:37:47.000000000 -0700
@@ -18,7 +18,7 @@
     )
 
   # C extension module
-  ctx(
+  shlibctx = ctx(
     features     = 'c cshlib pyext',
     source       = ['cairomodule.c',
                     'context.c',
@@ -34,6 +34,16 @@
     install_path = pycairoLibDir,
     )
 
+  # Remove all -arch flags; apparently we know better than waf what architecture
+  # we should be building for
+  for var in ['CFLAGS_PYEMBED', 'CFLAGS_PYEXT', 'CXXFLAGS_PYEMBED',
+    'CXXFLAGS_PYEXT', 'LINKFLAGS_PYEMBED', 'LINKFLAGS_PYEXT']:
+    newvar = []
+    for ix, arg in enumerate(shlibctx.env[var]):
+      if '-arch' not in (arg, shlibctx.env[var][ix - 1]):
+        newvar.append(arg)
+    shlibctx.env[var] = newvar
+
   # C API
   ctx.install_files(os.path.join(ctx.env['PREFIX'], 'include', 'pycairo'),
                     'pycairo.h')
