--- a/configure.ac	2018-12-23 13:37:14.000000000 -0800
+++ b/configure.ac	2019-05-19 07:26:48.000000000 -0700
@@ -1874,6 +1874,11 @@
             then
                 CFLAGS="${UNIVERSAL_ARCH_FLAGS} -isysroot ${UNIVERSALSDK} ${CFLAGS}"
                 LDFLAGS="${UNIVERSAL_ARCH_FLAGS} -isysroot ${UNIVERSALSDK} ${LDFLAGS}"
+  		# Issue #18075: the default maximum stack size (8MBytes) is too
+  		# small for the default recursion limit. Increase the stack size
+  		# to ensure that tests don't crash
+ 		LDFLAGS="-Wl,-stack_size,1000000 $(LDFLAGS)"
+
                 CPPFLAGS="-isysroot ${UNIVERSALSDK} ${CPPFLAGS}"
             else
                 CFLAGS="${UNIVERSAL_ARCH_FLAGS} ${CFLAGS}"
@@ -2698,11 +2703,6 @@
 	Darwin/*)
 		LINKFORSHARED="$extra_undefs -framework CoreFoundation"

-		# Issue #18075: the default maximum stack size (8MBytes) is too
-		# small for the default recursion limit. Increase the stack size
-		# to ensure that tests don't crash
-		LINKFORSHARED="-Wl,-stack_size,1000000 $LINKFORSHARED"
-
 		if test "$enable_framework"
 		then
 			LINKFORSHARED="$LINKFORSHARED "'$(PYTHONFRAMEWORKDIR)/Versions/$(VERSION)/$(PYTHONFRAMEWORK)'
