
--- a/Source/JavaScriptCore/API/JSBase.h	Sat Apr  9 23:48:36 2016
+++ b/Source/JavaScriptCore/API/JSBase.h	Thu Apr 27 13:48:12 2017
@@ -142,9 +142,9 @@
 /* Enable the Objective-C API for platforms with a modern runtime. */
 #if !defined(JSC_OBJC_API_ENABLED)
 #ifndef JSC_OBJC_API_AVAILABLE_MAC_OS_X_1080
-#define JSC_OBJC_API_ENABLED (defined(__clang__) && defined(__APPLE__) && ((defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1090 && !defined(__i386__)) || (defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE)))
+#define JSC_OBJC_API_ENABLED (defined(__clang__) && defined(__APPLE__) && ((defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1090 && !defined(__i386__)) && !PLATFORM(GTK) || (defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE)))
 #else
-#define JSC_OBJC_API_ENABLED (defined(__clang__) && defined(__APPLE__) && ((defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1080 && !defined(__i386__)) || (defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE)))
+#define JSC_OBJC_API_ENABLED (defined(__clang__) && defined(__APPLE__) && ((defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1080 && !defined(__i386__)) && !PLATFORM(GTK) || (defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE)))
 #endif
 #endif
 
--- a/Source/JavaScriptCore/API/ObjCCallbackFunction.h	Sat Apr  9 23:48:36 2016
+++ b/Source/JavaScriptCore/API/ObjCCallbackFunction.h	Tue Apr 25 15:31:47 2017
@@ -29,7 +29,7 @@
 
 #if JSC_OBJC_API_ENABLED
 
-#import <JavaScriptCore/JSCallbackFunction.h>
+#import "JSCallbackFunction.h"
 
 #if defined(__OBJC__)
 JSObjectRef objCCallbackFunctionForMethod(JSContext *, Class, Protocol *, BOOL isInstanceMethod, SEL, const char* types);

--- a/Source/WebCore/platform/gtk/RenderThemeGtk2.cpp	Sat Apr  9 23:48:37 2016
+++ b/Source/WebCore/platform/gtk/RenderThemeGtk2.cpp	Thu Apr 27 10:01:54 2017
@@ -40,6 +40,7 @@
 #include "PaintInfo.h"
 #include "PlatformContextCairo.h"
 #include "RenderElement.h"
+#include "RenderBox.h"
 #include "TextDirection.h"
 #include "UserAgentStyleSheets.h"
 #include "WidgetRenderingContext.h"
--- a/Source/JavaScriptCore/assembler/MacroAssembler.h~
+++ b/Source/JavaScriptCore/assembler/MacroAssembler.h
@@ -986,7 +986,7 @@ public:
         if (bitwise_cast<uint64_t>(value * 1.0) != bitwise_cast<uint64_t>(value))
             return shouldConsiderBlinding();
 
-        value = abs(value);
+        value = std::abs(value);
         // Only allow a limited set of fractional components
         double scaledValue = value * 8;
         if (scaledValue / 8 != value)
--- a/Source/autotools/FindDependencies.m4~
+++ b/Source/autotools/FindDependencies.m4
@@ -104,7 +104,7 @@ GLIB_GSETTINGS
 
 # TODO: use pkg-config (after CFLAGS in their .pc files are cleaned up).
 case "$host" in
-    *-*-darwin*)
+    *-*-foo*)
         UNICODE_CFLAGS="-I$srcdir/Source/JavaScriptCore/icu -I$srcdir/Source/WebCore/icu"
         UNICODE_LIBS="-licucore"
         ;;
