--- a/configure.ac	2021-03-15 16:19:23.000000000 -0600
+++ b/configure.ac	2021-03-15 16:19:00.000000000 -0600
@@ -409,7 +409,7 @@
   backend_immodules="$backend_immodules,quartz"
   GDK_WINDOWING="$GDK_WINDOWING
 #define GDK_WINDOWING_QUARTZ"
-  GDK_EXTRA_LIBS="$GDK_EXTRA_LIBS -Wl,-framework,Cocoa"
+  GDK_EXTRA_LIBS="$GDK_EXTRA_LIBS -Wl -framework Cocoa -framework Carbon"
   DISABLE_ON_QUARTZ='%'
   AM_CONDITIONAL(USE_QUARTZ, true)
   if test "x$quartz_relocation" = xyes; then



--- a/m4/intlmacosx.m4	2020-04-27 08:36:55.000000000 -0600
+++ b/m4/intlmacosx.m4	2021-03-15 16:48:35.000000000 -0600
@@ -21,7 +21,7 @@
   AC_CACHE_CHECK([for CFPreferencesCopyAppValue],
     [gt_cv_func_CFPreferencesCopyAppValue],
     [gt_save_LIBS="$LIBS"
-     LIBS="$LIBS -Wl,-framework -Wl,CoreFoundation"
+     LIBS="$LIBS -Wl,-framework CoreFoundation"
      AC_LINK_IFELSE(
        [AC_LANG_PROGRAM(
           [[#include <CoreFoundation/CFPreferences.h>]],
@@ -36,7 +36,7 @@
   dnl Check for API introduced in Mac OS X 10.3.
   AC_CACHE_CHECK([for CFLocaleCopyCurrent], [gt_cv_func_CFLocaleCopyCurrent],
     [gt_save_LIBS="$LIBS"
-     LIBS="$LIBS -Wl,-framework -Wl,CoreFoundation"
+     LIBS="$LIBS -Wl,-framework CoreFoundation"
      AC_LINK_IFELSE(
        [AC_LANG_PROGRAM(
           [[#include <CoreFoundation/CFLocale.h>]],
@@ -50,7 +50,7 @@
   fi
   INTL_MACOSX_LIBS=
   if test $gt_cv_func_CFPreferencesCopyAppValue = yes || test $gt_cv_func_CFLocaleCopyCurrent = yes; then
-    INTL_MACOSX_LIBS="-Wl,-framework -Wl,CoreFoundation"
+    INTL_MACOSX_LIBS="-Wl,-framework CoreFoundation"
   fi
   AC_SUBST([INTL_MACOSX_LIBS])
 ])
