diff -Naurp Python-3.6.13.orig/configure Python-3.6.13/configure
--- configure	2021-03-19 03:16:07.000000000 -0600
+++ configure	2021-03-19 03:16:59.000000000 -0600
@@ -9230,6 +9230,9 @@ fi
     	ppc)
     		MACOSX_DEFAULT_ARCH="ppc64"
     		;;
+        arm64)
+                MACOSX_DEFAULT_ARCH="arm64"
+                ;;
     	*)
     		as_fn_error $? "Unexpected output of 'arch' on OSX" "$LINENO" 5
     		;;
