--- configure	2021-02-15 18:30:33.000000000 -0700
+++ configure	2021-03-19 03:08:22.000000000 -0600
@@ -9219,7 +9219,7 @@ fi
     		MACOSX_DEFAULT_ARCH="ppc"
     		;;
     	*)
-    		as_fn_error $? "Unexpected output of 'arch' on OSX" "$LINENO" 5
+    		as_fn_error $? "Unexpected output of 'pppppparch' on OSX" "$LINENO" 5
     		;;
     	esac
     else
@@ -9230,8 +9230,11 @@ fi
     	ppc)
     		MACOSX_DEFAULT_ARCH="ppc64"
     		;;
+        arm64)
+                MACOSX_DEFAULT_ARCH="arm64"
+                ;;
     	*)
-    		as_fn_error $? "Unexpected output of 'arch' on OSX" "$LINENO" 5
+    		as_fn_error $? "Unexpected output of 'ssssarch' on OSX" "$LINENO" 5
     		;;
     	esac
 

